lib/src/firestore.dart

```diff
@@ -607,6 +640,17 @@ class GeoPoint {
   int get hashCode => hash2(latitude, longitude);
 }
 
+/// An immutable object representing an array of bytes.
+class Blob {
+  final Uint8List _data;
+
+  Blob(List<int> data) : _data = new Uint8List.fromList(data);
+  List<int> get data => _data;
+
+  Blob.fromUint8List(this._data);
+  Uint8List toUint8List() => _data;
```

Comment:
Effective Dart suggests in this case to use `asUint8List()` as it does not copy:
https://www.dartlang.org/guides/language/effective-dart/design#prefer-naming-a-method-to___-if-it-copies-the-objects-state-to-a-new-object

I'd probably go with leaving it `toUint8List` but make it return fresh copy of `Uint8List`. But renaming to `asUint8List` would be fine with me as well.

Your call.

Reply:
I guess in my case, I wanted the "as" feature to prevent creating yet another object (I don't know the cost of new Uint8List.fromList but I was doing it twice in some cases between 2 objects where having the original object would have been fine. I named it this way to (almost) match somehow the web interface: https://firebase.google.com/docs/reference/js/firebase.firestore.Blob

I will name it as then if you're ok